python_modules/libraries/dagster-fivetran/dagster_fivetran_tests/experimental/test_asset_specs.py

```diff
@@ -0,0 +1,16 @@
+import uuid
+from typing import Callable
+
+from dagster_fivetran import FivetranWorkspace
+
+
+def test_fetch_fivetran_workspace_data(workspace_data_api_mocks_fn: Callable) -> None:
+    api_key = uuid.uuid4().hex
+    api_secret = uuid.uuid4().hex
+
+    resource = FivetranWorkspace(api_key=api_key, api_secret=api_secret)
+
+    with workspace_data_api_mocks_fn(include_sync_endpoints=False):
```

Comment:
why do we not include sync_endpoints here? I'm not sure I follow. Feels like the API should still be able to return a valid response here and we should be asserting that it's not used, but maybe I'm missing something.

Reply:
responses.RequestsMock() raises an error if not all mocked responses/endpoints are used in the test - it's mainly to avoid duplicating code. 

There is a set of endpoints that are used to fetch the data, and another set of endpoints that are used to sync and poll the data. We will be using both to test sync and poll methods in subsequent PRs.

Reply:
Ah gotcha, that honestly makes me feel even more strongly about using a client fake as opposed to mocking. If we have to make these somewhat arbitrary booleans to get things to work across all tests I think that's pretty undesirable. When a new person comes to this code I think that will be really confusing.

I really think something like what we did for dlift could be powerful here, we basically built a "fake dbt cloud with jaffle shop" https://github.com/dagster-io/dagster/blob/0a0c77b42d8138e8703db3b94fa8381b86e16036/examples/experimental/dagster-dlift/dagster_dlift_tests/conftest.py#L47 and then all of the tests can just run against that thing. Kinda annoying to build, but test logic stays simple. What do you think?

Reply:
I updated the code to avoid using the boolean in [7408fa2](https://github.com/dagster-io/dagster/pull/25788/commits/7408fa23fa6626aa66ecfd855eeaa00480b0a978) - I agree that it was kinda awkward.

About the client fake, in this case, I think I see more value in mocking requests for 2 reasons. 

1. It makes the conftest file more straightforward. It also makes obvious that the only thing that we are mocking are the API requests and responses. 
2. I think the client fake solution is great, but it kinda adds a layer of complexity for external contributors that would like to contribute to the integration. Like you mentioned, it's more complex to build, so I would avoid using a solution that would be harder to update for contributors. Since mocking for unit tests is widely adopted by the Python community, it seems like the most logical strategy to adopt here.

Reply:
Yea it's tricky. For what it's worth, I don't think it being more complex to build _initially_ means it's more complex to build _on top_ of. I think the idea would be a client fake solution requires more up front effort, but for 90% of use cases doesn't require users to touch the API surface area at all - they just interact with the instance methods themselves.

I do see your point however around mocking kinda being the "devil we know" so to speak, regarding external contributors, though. It's complexity they're familiar with. 

I like the solution you landed on with the hierarchical API mocks more, feels more structured. Given that there's only a few APIs that we're mocking out here for all the tests, I think I'm OK with pushing this through with the mocking solution. Thanks for talking through it!